### PR TITLE
Miscellanous UI updates after testing and stakeholder feedback

### PR DESCRIPTION
**Pull request by mferrarini**
**Miscellanous UI updates after testing and stakeholder feedback**
* * *

**JIRA Tickets**:
* [HGL-345: Apply library design styles](https://jira.huit.harvard.edu/browse/HGL-345)
* [HGL-355: Enable "click to wake" function for map search](https://jira.huit.harvard.edu/browse/HGL-355)
* [HGL-416: Expand/collapse verbose metadata field](https://jira.huit.harvard.edu/browse/HGL-416)
* [HGL-463: Insert Google Analytics Tag script into HGL markup](https://jira.huit.harvard.edu/browse/HGL-463)

# What does this Pull Request do?
Incorporating UI updates after testing and stakeholder feedback to create a better user experience.

# How should this be tested?
The "test case" links are for environments, so be in the development_collapse-metadata branch

**Homepage**
* Search box
   * Updated placeholder text to say 'Try "population data" or "Cambridge"'
   * When click into the search box, placeholder text should disappear
   * Turned off autocomplete for search box, so when you're typing you shouldn't get a dropdown of titles/suggested searches
* "Click to wake" functionality added and appears on all maps; map initially appears with an opacity layer over it that disappears after clicking the button; scrolling off map will eventually cause it "sleep" again (we can also adjust how long the sleep/wake states are if we want something different than the default, [see options](https://github.com/harvard-huit/Harvard-Geospatial-Library/blob/9849b54defb6c6789bef7b2e7c76eaa8415f78c8/app/assets/javascripts/leaflet-sleep.js#L63-L72))
* Updated broken footer link for "GIS in the libraries (Canvas)" *note: the contact email is going to be updated with a new hgl_ref email once that is set up
* Google Analytics tag can be found in the document head (included at the bottom of the head element, look for 'Global site tag (gtag.js) - Google Analytics' commented out); can also be found in the [base.html.erb file](https://github.com/harvard-huit/Harvard-Geospatial-Library/commit/95a89e310c0ae2c9de61b33d5ffb61c953d8ffda)
* Started reworking the homepage layout, but final product will be in the development__homepage-revised branch, so other things can be ignored for now

**Search results page**
* Map sticks to top of page when scrolling through a long list of search results
* Clicking the "Clear all" link should remove all facets, but keep you on the search results page (before it was bouncing users back to the homepage)
* Filter rail + facets
   * Collapsed by default
   * When you select a facet, that category will remain open as the page reloads, but categories with unselected facets will collapse again
   * Facet links are displayed as block elements so all words will be underlined on hover (previous it was a flex element and lengthy facet links only had one big underline at the bottom of its flex box)
   * Restyled selected facets: bold text and "x" icon aligns with the top line for lengthy facets 

**Item detail page**
* Sidebar (discovered a few tweaks to the cards that needed to be made after more items were included in our test set)
   * New "Derived Datasets" card styled with same sizes and padding as other cards (test case: [2016 NYC Geodatabase, ArcGIS Version (jan2016)](https://localhost:3001/catalog/nyu_2451_34636)) *applied styles should now be generic enough to account for other cards that may appear up when we have more complete data
   * Padding added between buttons when there are multiple download buttons for an item (test case: [A new & accurate map of Asia](https://localhost:3001/catalog/57f0f116-b64e-4773-8684-96ba09afb549))
* Metatdata
   * "Place(s)" and "Subject(s)" now appear as one item per line (test case: [1-Meter Shaded Relief Multibeam Bathymetry Image](https://localhost:3001/catalog/stanford-dp018hs9766))
   * If there are more then 6 items, the field gets truncated and buttons appear to toggle more/less items (test case: [1:1,000,000-Scale National Wilderness Preservation System of the United States, 2014](https://localhost:3001/catalog/stanford-gd328br6434))
   * Metadata field simplified from "More details at" to "More details" (can view this with either test case above)
* Index map (test case: [Dabao Kinbōzu, Maps Index](https://localhost:3001/catalog/stanford-fb897vt9938))
   * Tray now slides down on first click (used to just quickly appear on the first click, but would slide down correctly after it was closed and reopened)
   * Reformatted table so that there's less space between the metadata titles and descriptions
   * Re-added website URL to the thumbnail image so that it's a clickable link (note: more to say about this and what i discovered in [HGL-374 ticket](https://jira.huit.harvard.edu/browse/HGL-374))
* Attribution table (test case: [2005 Rural Poverty GIS Database: Uganda](https://localhost:3001/catalog/stanford-cz128vq0535))
   * Re-added the scroll bar to account for long tables
   * Now behaves like the index map dropdown tray (has instructions with the pointer icon and the tray can be opened/closed at user interacts with the map)

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz
@phil-plencner-hl

### CODE REVIEW

File: app/views/catalog/_show_default_attribute_table.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<% if show_attribute_table? %>
+  <div class='attribute-table'>
+    <div class="attribute-table__directions">
+      Click map to inspect values
+      <span>
+        <svg width="16px" height="18px" viewBox="0 0 16 18" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+          <title>Pointer finger</title>
+          <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
```
Comment (phil-plencner-hl) [inline app/views/catalog/_show_default_attribute_table.html.erb:8]: Can we make our commonly used SVGs into partials?

File: app/views/layouts/blacklight/base.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    <title><%= render_page_title %></title>
+    <%= opensearch_description_tag application_name, opensearch_catalog_url(format: 'xml') %>
+    <%= favicon_link_tag %>
+    <%= stylesheet_link_tag "application", media: "all" %>
+    <%= javascript_include_tag "application" %>
+    <%= csrf_meta_tags %>
+    <%= content_for(:head) %>
+
+    <!-- Global site tag (gtag.js) - Google Analytics -->
```
Comment (phil-plencner-hl) [inline app/views/layouts/blacklight/base.html.erb:19]: Let's move the GA code into a partial and render is separately.

File: app/views/catalog/_show_default_attribute_table.html.erb (hunk 1 above)
Comment (mferrarini) [reply app/views/catalog/_show_default_attribute_table.html.erb]: Made a new folder to hold our svgs. Moved all our svgs into this folder and updated their calls in the appropriate erb files 

[https://github.com/harvard-huit/Harvard-Geospatial-Library/pull/51/commits/206881455302ca328efa095d0b25780020ef47ba](https://github.com/harvard-huit/Harvard-Geospatial-Library/pull/51/commits/206881455302ca328efa095d0b25780020ef47ba)

File: app/views/layouts/blacklight/base.html.erb (hunk 2 above)
Comment (mferrarini) [reply app/views/layouts/blacklight/base.html.erb]: GA now rendered separately and placed in the footer; removed base.html.erb file as we no longer have customizations to the base file
[https://github.com/harvard-huit/Harvard-Geospatial-Library/pull/51/commits/f5688c484b3ccd47b3e64281d1deed5164ce8cdb](https://github.com/harvard-huit/Harvard-Geospatial-Library/pull/51/commits/f5688c484b3ccd47b3e64281d1deed5164ce8cdb)

Comment (phil-plencner-hl) [reply app/views/layouts/blacklight/base.html.erb]: Moved back into base.html.erb, partial in place. looks good!

File: app/views/catalog/_show_default_attribute_table.html.erb (hunk 1 above)
Comment (phil-plencner-hl) [reply app/views/catalog/_show_default_attribute_table.html.erb]: Tested this out. looks great!

**Review by phil-plencner-hl — APPROVED** [review]
I have reviewed all the functionality and code. Works great and looks good!

**Pull request merged**